### PR TITLE
ncc build fixes

### PR DESCRIPTION
**Pull request by serhii-pavlovskyi-altran**
first commit unbreaks configure on system with `libclang-12-dev`, `clang-12`, `ninja` and `cmake 3.17+`(`ninja` says `multiple rules generate ncc_naming_style/main.cpp.ncc_style`). with lower `cmake` version configure succeeds, but build exits immediately with same error. by replacing `ninja` with `make` error becomes warning(it's still significant, `make` just skips duplicate rules, i.e. doesn't check style of some source files, rule duplication is genuine bug). without `libclang-12-dev` and `clang-12` `ENABLE_NCC_STYLE` is `OFF` and bug is not triggered

second commit silences fatal warning on current `gcc`
third commit fixes style issue uncovered by first commit
fourth and fifth commits fix build issues on current `gcc` outside of gpu plugin. only first three commits are required to build gpu plugin

there was another commit removing NullStatement check, but this change is already in master https://github.com/openvinotoolkit/openvino/pull/10254/commits/303060f91df8fa0d8501aec0c9eb56596b938433

### CODE REVIEW

**Review by apankratovantonp — APPROVED** [review]

File: cmake/developer_package/ncc_naming_style/ncc_naming_style.cmake [hunk 1]
```diff
@@ -107,8 +107,10 @@ function(ov_ncc_naming_style)
 
     list(APPEND NCC_STYLE_ADDITIONAL_INCLUDE_DIRECTORIES "${NCC_STYLE_SOURCE_DIRECTORY}")
 
+# without it sources with same name from different directories will map to same .ncc_style target
```
Comment (ilya-lavrenov) [inline cmake/developer_package/ncc_naming_style/ncc_naming_style.cmake:110]: please, align the comment

Comment (serhii-pavlovskyi-altran) [reply cmake/developer_package/ncc_naming_style/ncc_naming_style.cmake]: done

**Review by ilyachur — APPROVED** [review]

**Comment by ilyachur** [comment]
Please fix code style

**Comment by ilya-lavrenov** [comment]
@serhii-pavlovskyi-altran thanks for the fix!

**Pull request merged**